docs/contributing/core/index.md

```diff
@@ -0,0 +1,268 @@
+-   password: `admin`
+
+
+## Use the correct branch
+
+```{important}
+This section applies to members of the GitHub `plone/developers` team, who have write access to repositories under the Plone GitHub organization.
+
+Members of the `plone/contributors` team do not have write access, and instead must follow the process to set up their remote upstream and origin branches as described in {ref}`set-up-your-environment-label`.
+```
+
+The current default and development branch of `buildout.coredev` is `6.1`.
+Older versions are named according to their `major.minor` version.
+
+Always begin by checking out the git branch on which you want to work, usually called `origin`.
+If you have not yet checked out a branch, then you need to track it with the `-t` option and specify the remote branch to track.
+The following command will switch and track changes from the remote branch `origin/6.1`.
+
+```shell
+git checkout -t origin/6.1
+```
+
+If you check out a different branch, and want to return to the previous branch, you need to specify only the local branch.
+
+```shell
+git checkout 6.1
+```
+
+Next pull down and merge any recent changes from the remote tracked repository with a single command.
+
+```shell
+git pull
+```
+
+```{important}
+Make sure to rerun buildout for the current branch to get the correct versions of packages, otherwise you will get some weird behavior.
+```
+
+Next create a new branch on which you want to work from the current branch, tracking the upstream Plone repository, and check it out.
+It's a good idea to use a branch name that includes the issue number and is descriptive of what it resolves.
+
+```shell
+git switch -c 123-thing-i-fixed -t origin/6.1
+```
+
+Now you can edit your code without affecting the original branch.
+
+
+## Edit packages
+
+First identify the names of the Plone packages you want to work on.
+If you do not know, you can ask in the [Plone Community Forum](https://community.plone.org/).
+Only a few packages are in {file}`src/` by default.
+
+Edit the file {file}`checkouts.cfg` at the root of the repository by adding the package names under the `auto-checkout` list.
+
+```{code-block} cfg
+:emphasize-lines: 10-
+[buildout]
+always-checkout = force
+# always-checkout = false
+auto-checkout =
+# These packages are always checked out:
+    docs
+    # <snip>
+# These packages are manually added, or automatically added by mr.roboto:
+    # <snip>
+    plone.app.event
+    icalendar
+```
+
+Then rerun buildout to install the source packages to edit.
+
+```shell
+./bin/buildout
+```
+
+```{seealso}
+For tips on working with `mr.developer`, see {doc}`mrdeveloper`.
+```
+
+
+## Test locally
+
+If you change the expected behavior of a feature in a package, you should write a test to cover the change.
+
+To run a test for the specific package that you modified, use the `-m` option followed by the package name, as shown in the following example.
+
+```shell
+./bin/test -m plone.app.caching
+```
```

Comment:
We could add a note that `bin/test` is a script installed by buildout which runs the Zope testrunner. Newcomers may be expecting pytest.

Reply:
That belongs more in reference, or maybe explanation, than a how-to guide. Perhaps "Test framework of Plone core"?